src/fam.rs

```diff
@@ -938,4 +959,55 @@ mod tests {
         type Message2FamStructWrapper = FamStructWrapper<Message2>;
         assert!(serde_json::from_str::<Message2FamStructWrapper>(data_ser.as_str()).is_err());
     }
+
+    #[test]
+    fn test_clone_multiple_fields() {
+        #[derive(Default)]
+        #[repr(C)]
+        struct Foo {
+            index: u32,
+            length: u16,
+            flags: u32,
+            entries: __IncompleteArrayField<u32>,
+        }
+
+        generate_fam_struct_impl!(Foo, u32, entries, u16, length, 100);
+
+        type FooFamStructWrapper = FamStructWrapper<Foo>;
+
+        let mut foo = FooFamStructWrapper::new(0).unwrap();
+        foo.as_mut_fam_struct().index = 1;
+        foo.as_mut_fam_struct().flags = 2;
+        foo.as_mut_fam_struct().length = 3;
+        foo.push(3).unwrap();
+        foo.push(14).unwrap();
```

Comment:
if `length` is 3 shouldn't we push 3 elements?

Reply:
Well, this was confusing for me as well, when you use `push()` it appends the element at the end of the entries collection which is considered as already having 3 elements (because of the above line). Anyway, this is not something that was added in this PR and I should've probably kept the tests simpler. Also, the whole `fam` module needs better documentation :D.